e2e/assert.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```

Comment:
this script loads `bashunit` and allows to call the `assert*` bash-functions of `bashunit`.

I made it return 0 on successfull assertion or print the `bashunit` error on failure

<img width="569" alt="grafik" src="https://github.com/phpstan/phpstan-src/assets/120441/241012ad-6fc9-4ea8-8deb-f7f8d1c46fe8">

Reply:
if you like the approach I could adjust remaining shell assertions in the other test cases

Reply:
This is a pretty cool approach. I was thinking about doing something similar. Maybe, actually, we could have this code somehow inside bashunit itself and allow people using this feature if they would want to as well (inspired by your idea here https://github.com/phpstan/phpstan-src/pull/3160#discussion_r1640951032).

I will experiment a couple of hours in the train today, and I will give you an update later today 🏗️